micro-rdk/src/esp32/tls.rs

```diff
@@ -198,10 +198,10 @@ impl Esp32TLS {
                 servercert_bytes: cfg.srv_cert.len() as u32,
             },
             __bindgen_anon_5: crate::esp32::esp_idf_svc::sys::esp_tls_cfg_server__bindgen_ty_5 {
-                serverkey_buf: cfg.srv_key,
+                serverkey_buf: cfg.srv_key.as_ptr(),
```

Comment:
Above on 195 that `as_ptr` created, I believe, a use-after-free. Nothing is holding alive the state in `Esp32TLSServerConfig` such that these pointers are valid once we get into `serve`.